ls1intum/Artemis
# Make buttons work like links

### Checklist

### Motivation and Context

Closes #611.

### Description

This PR basically changes all buttons which were using (router) links to be links looking like buttons using a bit of CSS and built-ins (it is actually recommended to use links instead of buttons). Now most buttons work properly with middle mouse/right clicks. Buttons I explicitly did not change are `Save`, `Cancel`, `Delete`, etc. and buttons opening modals, since it does not make sense for those to open in a new tab.
There are two exceptions to this:
- Buttons in the students view
- Following buttons for programming exercises: "Generate new Programming Exercise", "Edit in Editor", "Check Plagiarism", "Grading", "Online editor"

I did not change those since they require a bit more work. This PR is already big and will be time consuming to review, so delaying those changes to a later PR where we can focus on having the student-view regression-free is a good idea. Another follow-up will expand our client documentation.

### Steps for Testing

Testing is not hard but time consuming. In principle middle-mouse click on every button. Those buttons which you'd expect to work as links should open a new tab (e.g. `Lectures`), and those you think shouldn't work also shouldn't open anything. (= Please give me feedback in case a button works unexpected.) Again, buttons that open modals, and `Save`, `Cancel`, `Delete`, etc. should explicitly not work as links.
Some things to pay extra attention to: The create and edit routes. When opening those in a new tab make sure saving and cancelling works and redirects you to the detail view (when editing) or overview (when creating).
Please also test the buttons in exams.

### Detailed steps for testing

Everything in quotation marks is a button that has to be tested. Every button not listed here should not work.

Server Administration:
- User Management: "Create" (which takes takes you back to the overview when finished), "View", "Edit" (which takes you back to the single view when finished)
- Single user view: "Edit" (which takes you back to the single view when finished)
- Notification Management: "Create" (which takes you back to the overview when finished), "View", "Edit" (whiche takes you back to the single view when finished)
- Single notification view: "Edit" (which takes you back to the single view when finished)

Course Management:
- "Create new course" (which takes you back to the overview when finished)
- For a single course (both overview and single detail view): "Exercises", "Scores", "Assessment Dashboard", "Instructor Dashboard", "Lectures", "Exams", "Edit" (which takes you back to the single view when finished)
- Additionally on the single course detail view: "Learning goals" and "Q&A"
- Lectures (single and overview): "Create", "Edit", "Attachments", "Units"
- Lecture units: All four "Create" buttons, "Edit" buttons for text, attachments, videos
- Assessment & Instructor Dashboards: "Exercise Dashboard", "Submissions"

Exercises (part of course management):
- "Create": drag and drop quiz, quiz, modeling exercise, text exercise, file upload exercise (which take you back to the overview page when finished)
- Drag and drop quizzes/Apollon diagrams: "Open" (note that the Save button on that page and the create page does not take you back!)
- For modeling exercises, text exercises and file upload exercises:
  - On the overview: "View", "Edit" (takes you back to the detail page when finishing), "Scores", "Submissions", "Participations"
  - The "Edit" button on the detais page (which takes you back to the detail page when finished)
  - The "Check Plagiarism" button on the detais page
  - The "Participations" and "View Assessment" button on the "Scores" page
  - Assessments/Submissions: "Assess Model"
- For quiz exercises:
  - On the overview: "Preview", "Sample Solution", "Statistics", "Re-evaluate", "Edit" (which takes you back to the exercises overview when finished)
  - The "Participations" and "View Assessment" button on the Scores page

Exams (part of course management):
- "Create" (sends you back to the overview when finished)
- "Assessment Dashboard", "Scores", "Exercise Groups", "Test Run", "Student Exams", "Students", "Edit" (which takes you back to the detail page once finished)
- On the Assessment Dashboard: "Exercise Dashboard"
- On the Exercise Groups overview:
  - "Add" for quiz, text, programming and modeling exercises (which takes you back to the overview when finished) 
  - The rest of the buttons on that page are the same as on the exercises overview of a course and have to behave the same way (except taking you back to the groups overview when done editing one)
- On the Test Runs page: "Summary"
- On the Student Exams page: "View"
- On the details page of a single exam: "Student Exams", "Edit" (which takes you back to the details page when finished)
